ignite/config/plugins/persist.go

```diff
@@ -0,0 +1,37 @@
+package plugins
+
+import (
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+// persist writes a plugin configuration file to a specified file.
+// the configuration state that is passed in will be the new state of the file
+// before writing the new definition to disk, a truncatte and seek operation
+// are performed to assure the file contnets will be overriden.
+func persist(config *Config, path string) error {
+	f, err := os.OpenFile(path, os.O_RDWR, 0o644)
+	if err != nil {
+		return err
+	}
+
+	// zero out the file for a new config write
+	err = f.Truncate(0)
+	if err != nil {
+		return err
+	}
+	// seek to the beginning to assure there are no trailing properties
+	_, err = f.Seek(0, 0)
+	if err != nil {
+		return err
+	}
+
+	// check that plugins are in fact defined within the config
+	// if there is an empty array of plugins the encoding will be `{}`
```

Comment:
This is because of the `omitempty` tag option in `Config.Plugins` field. If we remove it, an empty config is then serialized into:

```yaml
plugins: []
```

Which is fine I think. So let's remove the `omitempty`, which had sense in the context of the chain config file, but not anymore. After that you can simplify this function.